src/Microsoft.Extensions.SecretManager.Tools/Internal/DotNetCoreSdkResolver.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Extensions.ProjectModel.Internal
+{
+    internal class DotNetCoreSdkResolver
+    {
+        private readonly string _installationPath;
+
+        public DotNetCoreSdkResolver(string installationDir)
+        {
+            _installationPath = installationDir;
+        }
+
+        private IEnumerable<string> Installed
+            => Directory.EnumerateDirectories(Path.Combine(_installationPath, "sdk"));
+
+        private string ResolveGlobalJsonSdkVersion(string start)
+        {
+            var dir = new DirectoryInfo(start);
+            FileInfo globalJson = null;
+            while (dir != null)
+            {
+                var candidate = Path.Combine(dir.FullName, "global.json");
+                if (File.Exists(candidate))
+                {
+                    globalJson = new FileInfo(candidate);
+                    break;
+                }
+                dir = dir.Parent;
+            }
+            if (globalJson == null)
+            {
+                return null;
+            }
+            var t = new { sdk = new { version = "" } };
+            try
+            {
+                var obj = JsonConvert.DeserializeAnonymousType(File.ReadAllText(globalJson.FullName), t);
+                return obj?.sdk?.version;
+            }
+            catch (JsonException)
+            {
+                // TODO log
+                return null;
+            }
+        }
+
+        public DotNetCoreSdk ResolveLatest()
+        {
+            var first = Installed.Select(d => new { path = d, version = SemanticVersion.Parse(Path.GetFileName(d)) })
+                .OrderByDescending(sdk => sdk.version)
+                .First();
```

Comment:
In theory, it's not possible for "dotnet user-secrets" to ever be invoked without some SDK installed...but at the very least we need a better error message for that.